pkg/vfs/archive.go

```diff
@@ -0,0 +1,62 @@
+package vfs
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+)
+
+// ZipMime is the content-type for zip archives
+const ZipMime = "application/zip"
+
+// Archive is the data to create a zip archive
+type Archive struct {
+	Name  string   `json:"name"`
+	Files []string `json:"files"`
+}
+
+// Serve creates on the fly the zip archive and streams in a http response
+func (a *Archive) Serve(c Context, w http.ResponseWriter) error {
+	header := w.Header()
+	header.Set("Content-Type", ZipMime)
+	header.Set("Content-Disposition", "attachment; filename="+a.Name+".zip")
```

Comment:
Apparently [it does](https://github.com/golang/go/blob/master/src/net/http/header.go#L155) so it seems fine. Although, we should probably disallow [newlines in filenames](https://github.com/cozy/cozy-stack/blob/master/pkg/vfs/vfs.go#L47)...